src/app/common/stores/applications/SsvWeb/Wallet.store.ts

```diff
@@ -133,7 +133,17 @@ class WalletStore extends BaseStore implements Wallet {
    */
   async connectWalletFromCache() {
     const selectedWallet: any = window.localStorage.getItem('selectedWallet');
-    if (selectedWallet && selectedWallet !== 'undefined') {
+    const walletConnectCondition = selectedWallet === 'WalletConnect' && !!window.localStorage.getItem('walletconnect');
+    const metamaskCondition = selectedWallet && selectedWallet !== 'undefined';
+    if (selectedWallet ===  'WalletConnect') {
+      await this.connectWallet(walletConnectCondition, selectedWallet);
+    } else {
+      await this.connectWallet(metamaskCondition, selectedWallet);
+    }
+  }
+
+  async connectWallet(condition: boolean, selectedWallet: any) {
```

Comment:
As I see - `condition` parameter is used only to identify if the flow should select and check wallet, otherwise there should not be any wallet usage in entire the app.

Please refactor the code in order to:
1. Rename `condition` to `useWallet`
2. For both wallet connect or metamask conditions - use additional method which says: `getWalletInfo()` and it should return `string` or `null`. If not null - then do wallet select, if null - then do `else` block.

Reply:
done